src/ConnectionPool.jl

```diff
@@ -496,16 +505,21 @@ function getconnection(::Type{SSLContext},
         sslconfig = global_sslconfig(require_ssl_verification)
     end
 
-    port = isempty(port) ? "443" : port
-    @debug 2 "SSL connect: $host:$port..."
     io = SSLContext()
     setup!(io, sslconfig)
-    associate!(io, getconnection(TCPSocket, host, port; kw...))
+    associate!(io, tcp)
     hostname!(io, host)
     handshake!(io)
     return io
 end
 
+function sslupgrade(t::Transaction{TCPSocket},
+                    host::AbstractString; kw...)::Transaction{SSLContext}
+    tls = sslconnection(t.c.io, host; kw...)
+    c = Connection(tls)
+    return client_transaction(c)
+end
```

Comment:
We reach down through the `Transaction` and the `Connection` to get to the raw TCP socket. (Same as what is done when upgrading to WebSockets).
We then create an SSLContext by calling `sslconnection`.
Since the caller was working with a `Transaction` object, they need the return value to be a `Transaction` object as well, so we wrap the new SSLContext in a `Connection`, then in an `Transaction`. At this point it looks exactly like a real HTTPS connection that might have come from the pool. Except that it is not in the pool. It is intended to be used once then GCed.